docs/designs/elasticdl_operator.md

```diff
@@ -0,0 +1,210 @@
+      --evaluation_throttle_secs '0' --checkpoint_dir_for_init '' --sync_version_tolerance
+      '0' --log_loss_steps '100' --use_async 'False' --lr_staleness_modulation 'False'
+    command:
+    - /bin/bash
+    env:
+    - name: MY_POD_IP
+      valueFrom:
+        fieldRef:
+          fieldPath: status.podIP
+    image: elasticdl:test
+    imagePullPolicy: Never
+    name: elasticdl-test-mnist-master
+    resources:
+      limits:
+        cpu: '1'
+        memory: 2048Mi
+      requests:
+        cpu: '0.2'
+        memory: 1024Mi
+    volumeMounts:
+    - mountPath: /data
+      name: elasticdl-test-mnist-master-volume-0
+  priorityClassName: ''
+  restartPolicy: Never
+  volumes:
+  - hostPath:
+      path: /data
+    name: elasticdl-test-mnist-master-volume-0
+```
+
+We could rewrite it as a custom ElasticDLJob object
+after the ElasticDL CRD is created.
+The following is a demo:
+
+```yaml
+apiVersion: "elasticdl.org/v1"
+kind: "ElasticAI"
+metadata:
+  name: "test-mnist"
+spec:
+  trainingArgs:
+    modelZoo: /model_zoo
+    modelDef: mnist_functional_api.mnist_functional_api.custom_model
+    trainingData: /data/mnist/train
+    validationData: /data/mnist/val
+    output: /data/output
+    dataReaderParams: ""
+    minibatchSize: 64
+    numMinibatchesPerTask: 20
+    evaluationStep: 1000
+    checkpointStep: 1000
+    checkpointDir: /data/checkpoint
+    checkpointDirForInit: ""
+    keepCheckpointMax: 0
+  Master:
+    template:
+      spec:
+        restartPolicy: Never
+        priorityClassName: high
+        containers:
+        - name: test-mnist-master
+          image: elasticdl-mnist
+          imagePullPolicy: Never
+          resources:
+            limits:
+              cpu: 1
+              memory: 2048Mi
+            requests:
+              cpu: 0.2
+              memory: 1024Mi
+          volumeMounts:
+          - mountPath: /data
+            name: test-volume
+        volumes:
+        - name: test-volume
+          hostPath:
+            path: /data
+  PS:
+    replicas: 2
+    template:
+      spec:
+        restartPolicy: Never
+        priorityClassName: high
+        containers:
+        - name: test-mnist-ps
+          image: elasticdl-ps
+          imagePullPolicy: Never
+          resources:
+            limits:
+              cpu: 1
+              memory: 2048Mi
+            requests:
+              cpu: 0.2
+              memory: 1024Mi
+  Worker:
+    replicas: 10
+    template:
+      spec:
+        restartPolicy: Never
+        priorityClassName: low
+        containers:
+        - name: test-mnist-worker
+          image: elasticdl-mnist
+          imagePullPolicy: Never
+          resources:
+            limits:
+              cpu: 1
+              memory: 2048Mi
+            requests:
+              cpu: 0.2
+              memory: 1024Mi
+          volumeMounts:
+          - mountPath: /data
+            name: test-volume
+        volumes:
```

Comment:
I am thinking how to convert this nested volume/volumeMounts data into worker arguments.
Currently, we are using something like `--volume='claim_name=glusterfs-directory,mount_path=/data'`.

Reply:
ElasticDL Controller will get the YAML file. It's just some key-value pairs, I think it's not hard to convert it inside the controller.